Adds release announcement post CLI for Microsoft 365 v6.8 (#1344)
pnp/blog: content/cli-for-microsoft-365/cli-for-microsoft-365-v6-8/index.md

---
title: CLI for Microsoft 365 v6.8
date: 2023-05-31T08:00:00.000Z
author: Martin Lingstuyl
githubname: martinlingstuyl
categories:
  - CLI for Microsoft 365
images:
  - images/blog/cli-for-microsoft365/banner-cli-m365.png
tags:
  - CLI for Microsoft 365
  - SharePoint
  - SharePoint Framework (SPFx)
type: popular
---

We've published a new minor version of CLI for Microsoft 365. 

> [CLI for Microsoft 365](https://aka.ms/cli-m365) is a cross-platform command-line tool. It helps you manage your Microsoft 365 tenant and SharePoint Framework projects. No matter which operating system or shell you use. This release is again brimming with new features and enhancements. We've added no less than **18 new commands** for managing multiple Microsoft 365. We've also added **40+ enhancements** to existing commands. This is also the release in which we added 🌶️ chili, our AI-based docs assistant and built an entire new site for the CLI for Microsoft 365. Interested? Read on for more details.

Author: ```suggestion
[CLI for Microsoft 365](https://aka.ms/cli-m365) is a cross-platform command-line tool. It helps you manage your Microsoft 365 tenant and SharePoint Framework projects. No matter which operating system or shell you use. This release is again brimming with new features and enhancements. We've added no less than **18 new commands** for managing multiple Microsoft 365 services. We've also added **40+ enhancements** to existing commands. This is also the release in which we added 🌶️ chili, our AI-based docs assistant and built an entire new site for the CLI for Microsoft 365. Interested? Read on for more details.
```

> Read the [release notes](https://aka.ms/cli-m365/notes) in full for all new features and improvements.
 
## What's new

### 🌶️ Chili, our AI-based docs assistant

Context-switching is sometimes called one of the big productivity killers. And yet, working with a commandline interface will often make us want to open the project website or search the web in order to find out how something works. We recently added an AI-based search engine to our documentation site to make searching our site better. This AI-assistent is powered by Mendable, a chat-based search tool. In this release we bring the AI-based search engine to the CLI. You can now get answers on specific questions without leaving the terminal.

Ask chili a question by running:

```sh
m365? "how can I upgrade my SPFx project?"
```

Or just type `m365?` and hit enter.

![cli-assistent](./images/cli-assistent.gif)

- [CLI assistent (chili)](https://pnp.github.io/cli-microsoft365/user-guide/chili/)
- [Check out **Mendable**](https://www.mendable.ai/)
 
### Setting up the CLI

> Some people use the CLI for Microsoft 365 interactively. Other people use it to build scripts that can run automomously. The new `m365 setup` command is a wizard-style command that helps you configure the CLI for Microsoft 365 for your needs. It will ask you a series of questions and based on your answers, it will configure the CLI for Microsoft 365 for you. It also has a few flags to quickly configure the CLI according to some presets based on best practices.

Reviewer: ```suggestion
Some people use the CLI for Microsoft 365 interactively. Other people use it to build scripts that can run autonomously. The new `m365 setup` command is a wizard-style command that helps you configure the CLI for Microsoft 365 for your needs. It will ask you a series of questions and based on your answers, it will configure the CLI for Microsoft 365 for you. It also has a few flags to quickly configure the CLI according to some presets based on best practices.
```

```sh
m365 pa app consent set --environment '4be50206-9576-4237-8b17-38d8aadfaa36' --name '3989cb59-ce1a-4a5c-bb78-257c5c39381d' --bypass true
```

- [pa app consent set](https://pnp.github.io/cli-microsoft365/cmd/pa/app/app-consent-set/)

### Working with Purview Information Protection

In this release we started implementing some commands for working with sensitivity labels in Microsoft Purview. 

To get a list of sensitivity labels, execute:

```sh
m365 purview sensitivitylabel list
```

To get the details of a specific sensitivity label, run:

```sh
m365 purview sensitivitylabel get --id 6f4fb2db-ecf4-4279-94ba-23d059bf157e
```

To retrieve policy settings for sensitivity labels, run:

```sh
m365 purview sensitivitylabel policysettings list
```

- [purview sensitivitylabel list](https://pnp.github.io/cli-microsoft365/cmd/purview/sensitivitylabel/sensitivitylabel-list/)
- [purview sensitivitylabel get](https://pnp.github.io/cli-microsoft365/cmd/purview/sensitivitylabel/sensitivitylabel-get/)
- [purview sensitivitylabel policysettings list](https://pnp.github.io/cli-microsoft365/cmd/purview/sensitivitylabel/sensitivitylabel-policysettings-list//)

### Managing SharePoint

> We've added more commands for working with SharePoint. We've made it easier to manage Application Customizers, ListView Command Sets and items in the recycle bin. You can also update large amounts of list items.

Reviewer 2: ```suggestion
We've added more commands for working with SharePoint. We've made it easier to manage Application Customizers, ListView Command Sets, and items in the recycle bin. You can also update large amounts of list items.
```

To remove an Application Customizer from a site, execute:

```sh
m365 spo applicationcustomizer remove --title "Some customizer" --webUrl https://contoso.sharepoint.com/sites/sales
```

To get a list of Application Customizers that are deployed tenant-wide, run:

```sh
m365 spo tenant applicationcustomizer list
```

To remove an Application Customizer that is deployed tenant-wide, run:

```sh
m365 spo tenant applicationcustomizer remove --title "Some customizer"
```

To remove a ListView Command Set that is deployed tenant-wide, run:

```sh
m365 spo tenant commandset remove --title "Some command set"
```

To update the properties of a ListView Command Set that is deployed tenant-wide, run:

```sh

> m365 spo tenant commandset  set --id 3  --clientSideComponentProperties '{ "someProperty": "Some value" }'

Reviewer 2: ```suggestion
m365 spo tenant commandset set --id 3  --clientSideComponentProperties '{ "someProperty": "Some value" }'
```

```

> To update a lot of items as quick as possible, run:

Reviewer 2: ```suggestion
To update a lot of items as quickly as possible, run:
```

```sh
m365 spo listitem batch set --filePath "C:\Path\To\Csv\CsvFile.csv" --webUrl https://contoso.sharepoint.com/sites/project-x --listTitle "Demo List"
```

> Remove an item from first-stage recycle bin:

Reviewer 2: This command is capable of removing items in both first stage and second-stage recycle bins.

Author: So what do you mean @milanholemans? You'd like an extra example for the second-stage scenario?

Reviewer 2: No just the description. Right now you are saying that this command can remove first-stage recycle bin items. But it can remove both first stage and second stage recycle bin items.

Author: ah, I see 😊

- [spo tenant commandset remove](https://pnp.github.io/cli-microsoft365/cmd/spo/tenant/tenant-commandset-remove/)
- [spo tenant commandset set](https://pnp.github.io/cli-microsoft365/cmd/spo/tenant/tenant-commandset-set/)

### Working with Microsoft Teams

Several additional commands have been made available for Teams. You can now remove members from a group chat and get a list of meeting transcripts.

To remove a member from a Teams chat, run:

```sh
m365 teams chat member remove --chatId 19:8b081ef6-4792-4def-b2c9-c363a1bf41d5_5031bb31-22c0-4f6f-9f73-91d34ab2b32d@unq.gbl.spaces --userName john.doe@contoso.com
```

To get meeting transcripts for a meeting, run:

```sh
m365 teams meeting transcript list --meetingId MSo1N2Y5ZGFjYy03MWJmLTQ3NDMtYjQxMy01M2EdFGkdRWHJlQ
```

- [teams chat member remove](https://pnp.github.io/cli-microsoft365/cmd/teams/chat/chat-member-remove/)
- [teams meeting transcript list](https://pnp.github.io/cli-microsoft365/cmd/teams/meeting/meeting-transcript-list/)

## What's changed

### Working with the SharePoint Term Store

> We've shipped several enhancements that are very useful when working the SharePoint Term Store. You no longer need to be a SharePoint administrator to manage the term store. If you only have permissions as a term store administrator, you can manage terms, term sets and term groups in the global term store using the CLI for Microsoft 365. Aside from the global term store you can now also manage term stores on site collections. For both scenario's you'll need to use the new `--webUrl` option that's added to every command. The CLI for Microsoft 365 will now connect to the global or local term store, using the site you've specified. 

Author: on-site is not what I meant though...

We can change it to `site-level term store`

> We've shipped several enhancements that are very useful when working the SharePoint Term Store. You no longer need to be a SharePoint administrator to manage the term store. If you only have permissions as a term store administrator, you can manage terms, term sets and term groups in the global term store using the CLI for Microsoft 365. Aside from the global term store you can now also manage term stores on site collections. For both scenario's you'll need to use the new `--webUrl` option that's added to every command. The CLI for Microsoft 365 will now connect to the global or local term store, using the site you've specified. 

Reviewer 2: ```suggestion
We've shipped several enhancements that are very useful when working with the SharePoint Term Store. You no longer need to be a SharePoint administrator to manage the term store. If you only have permissions as a term store administrator, you can manage terms, term sets, and term groups in the global term store using the CLI for Microsoft 365. Aside from the global term store, you can now also manage term stores on-site collections. For both scenarios you'll need to use the new `--webUrl` option that's added to every command. The CLI for Microsoft 365 will now connect to the global or local term store, using the site you've specified. 
```

To get a list of terms of a global term set, without being a SharePoint admin, run:

```sh
m365 spo term list --termGroupName "DMS" --termSetName "Business process" --webUrl "https://contoso.sharepoint.com/sites/project-x"
```

To get a list of term sets of a global term group, without being a SharePoint admin, run:

```sh

> m365 spo term set list --termGroupName "DMS" --webUrl https://contoso.sharepoint.com/sites/project-x

Author: ```suggestion
m365 spo term set list --termGroupName "DMS" --webUrl "https://contoso.sharepoint.com/sites/project-x"
```

```

To get a list of terms of a local (site-specific) term set, run:

```sh
m365 spo term list --termGroupName "Sitecollection - contoso.sharepoint.com-sites-Project-x" --termSetName "Themes" --webUrl "https://contoso.sharepoint.com/sites/project-x"
```

The following commands were impacted:

- [spo term add](https://pnp.github.io/cli-microsoft365/cmd/spo/term/term-add/)
- [spo term get](https://pnp.github.io/cli-microsoft365/cmd/spo/term/term-add/)
- [spo term list](https://pnp.github.io/cli-microsoft365/cmd/spo/term/term-list/)
- [spo term group get](https://pnp.github.io/cli-microsoft365/cmd/spo/term/term-group-get/)
- [spo term group list](https://pnp.github.io/cli-microsoft365/cmd/spo/term/term-group-list/)
- [spo term set add](https://pnp.github.io/cli-microsoft365/cmd/spo/term/term-set-add/)
- [spo term set get](https://pnp.github.io/cli-microsoft365/cmd/spo/term/term-set-get/)
- [spo term set list](https://pnp.github.io/cli-microsoft365/cmd/spo/term/term-set-list/)

### Updating Azure AD Users

> You could already update user information for Azure AD users. In the previous release we added a lot of well-known options, like job title and office location. We've now extended this to any property, even if it's unknown to the CLI for Microsoft 365.

Reviewer: ```suggestion
You could already update user information for Azure AD users. In the previous release, we added a lot of well-known options, like job title and office location. We've now extended this to any property, even if it's unknown to the CLI for Microsoft 365.
```

> You could already update user information for Azure AD users. In the previous release we added a lot of well-known options, like job title and office location. We've now extended this to any property, even if it's unknown to the CLI for Microsoft 365.

Author: ```suggestion
You could already update user information for Azure AD users. In the previous release we added a lot of well-known options, like 'job title' and 'office location'. We've now extended this to any property, even if it's unknown to the CLI for Microsoft 365.
```

> You could already update user information for Azure AD users. In the previous release we added a lot of well-known options, like job title and office location. We've now extended this to any property, even if it's unknown to the CLI for Microsoft 365.

Reviewer 2: ```suggestion
You could already update user information for Azure AD users. In the previous release, we added a lot of well-known options, like job titles and office locations. We've now extended this to any property, even if it's unknown to the CLI for Microsoft 365.
```

To update the employee hire date of a user, run:

```sh
m365 aad user set --userPrincipalName "adele.vance@contoso.com" --employeeHireDate "2022-01-01"
```

To create a new user and set the employee hire date, run:

```sh
m365 aad user add --displayName "Adele Vance" --userName "adele.vance@contoso.com" --password "SomePassw0rd" --companyName Contoso --department Sales --forceChangePasswordNextSignIn --employeeHireDate "2023-05-31"
```

- [aad user add](https://pnp.github.io/cli-microsoft365/cmd/aad/user/user-add/)
- [aad user set](https://pnp.github.io/cli-microsoft365/cmd/aad/user/user-set/)

### 🌍 One more thing: We've launched a new website!!

> Updating our website was a thing we wanted to do for quite some time now. We've swapped to using a new static site generator: Docusaurus. It's been a major thing in terms of development time and discussions internally. But if we've done our job well, the changes should be minimal. 🤞 Of course, you may notice some styling changes... 😊some things have shifted slightly. We also temporarily dropped the search bar. (We'll get it back in due time!) But in the meantime we fortunately have our AI-assistent, so you can still search for answers to your questions.

Reviewer: ```suggestion
Updating our website was a thing we wanted to do for quite some time now. We've swapped to using a new static site generator: Docusaurus. It's been a major thing in terms of development time and discussions internally. But if we've done our job well, the changes should be minimal. 🤞 Of course, you may notice some styling changes... 😊some things have shifted slightly. We also temporarily dropped the search bar. (We'll get it back in due time!) But in the meantime, we fortunately have our AI assistant, so you can still search for answers to your questions.
```

> Updating our website was a thing we wanted to do for quite some time now. We've swapped to using a new static site generator: Docusaurus. It's been a major thing in terms of development time and discussions internally. But if we've done our job well, the changes should be minimal. 🤞 Of course, you may notice some styling changes... 😊some things have shifted slightly. We also temporarily dropped the search bar. (We'll get it back in due time!) But in the meantime we fortunately have our AI-assistent, so you can still search for answers to your questions.

Reviewer 2: ```suggestion
Updating our website was a thing we wanted to do for quite some time now. We've swapped to using a new static site generator: Docusaurus. It's been a major thing in terms of development time and discussions internally. But if we've done our job well, the changes should be minimal. 🤞 Of course, you may notice some styling changes... 😊some things have shifted slightly. We also temporarily dropped the search bar. (We'll get it back in due time!) But in the meantime, we fortunately have our AI-assistant, so you can still search for answers to your questions.
```

- [Adam Wójcik](https://github.com/Adam-it)
- [Ganesh Sanap](https://github.com/ganesh-sanap)
- [Jasey Waegebaert](https://github.com/Jwaegebaert)
- [Martin Lingstuyl](https://github.com/martinlingstuyl)
- [Mathijs Verbeeck](https://github.com/MathijsVerbeeck)
- [Milan Holemans](https://github.com/milanholemans)
- [Nanddeep Nachan](https://github.com/nanddeepn)
- [Nico De Cleyre](https://github.com/nicodecleyre)
- [Nicolas Camara](https://github.com/nickscamara)
- [Saurabh K. Tripathi](https://github.com/Saurabh7019)
- [Waldek Mastykarz](https://github.com/waldekmastykarz)

Thank you all for the time you chose to spend on CLI for Microsoft 365 and for your help to advance it!

### High fives

CLI for Microsoft 365 wouldn't be where it is today if it weren't for our users who provide us with feedback. High fives to the following people who took the time to share their feedback and ideas for improvement with us (in alphabetical order):

- [Dakota Wray](https://github.com/DakotaWray2)
- [doronyaary](https://github.com/doronyaary)
- [Joshua Probst](https://github.com/joshua-probst)
- [Siddharth Vaghasia](https://github.com/siddharth-vaghasia)

> 

Reviewer 2: Aren't we missing [dojcsakj](https://github.com/dojcsakj)?
https://github.com/pnp/cli-microsoft365/issues/4865

Author: I haven't yet updated these lists... ran out of time :-)

Author: @milanholemans, do you remember what the decision was about the high fives? Do we list everyone? or exclude maintainers and contributors?

Reviewer 2: As far as I know, we exclude maintainers and contributors.

Author: That seems wise. I'll update it :-)
